Psychtoolbox/PsychHardware/EyelinkToolbox/EyelinkBasic/PsychEyelinkDispatchCallback.m

```diff
@@ -129,326 +161,419 @@
-        f=el.drift_correction_target_beep(1);
-        v=el.drift_correction_target_beep(2);
-        d=el.drift_correction_target_beep(3);
-    case 'calibration_failed_beep'
-        doBeep=el.feedbackbeep;
-        f=el.calibration_failed_beep(1);
-        v=el.calibration_failed_beep(2);
-        d=el.calibration_failed_beep(3);
-    case 'calibration_success_beep'
-        doBeep=el.feedbackbeep;
-        f=el.calibration_success_beep(1);
-        v=el.calibration_success_beep(2);
-        d=el.calibration_success_beep(3);
-    case 'drift_correction_failed_beep'
-        doBeep=el.feedbackbeep;
-        f=el.drift_correction_failed_beep(1);
-        v=el.drift_correction_failed_beep(2);
-        d=el.drift_correction_failed_beep(3);
-    case 'drift_correction_success_beep'
-        doBeep=el.feedbackbeep;
-        f=el.drift_correction_success_beep(1);
-        v=el.drift_correction_success_beep(2);
-        d=el.drift_correction_success_beep(3);
-    otherwise
-        % some defaults
-        doBeep=el.feedbackbeep;
-        f=500;
-        v=0.5;
-        d=1.5;
-end
 
-if doBeep==1
-    Beeper(f, v, d);
+    function EyelinkMakeSound(el, s)
+        % set all sounds in one place, sound params defined in
+        % eyelinkInitDefaults
+        switch(s)
+            case 'cal_target_beep'
+                doBeep=el.targetbeep;
+                f=el.cal_target_beep(1);
+                v=el.cal_target_beep(2);
+                d=el.cal_target_beep(3);
+            case 'drift_correction_target_beep'
+                doBeep=el.targetbeep;
+                f=el.drift_correction_target_beep(1);
+                v=el.drift_correction_target_beep(2);
+                d=el.drift_correction_target_beep(3);
+            case 'calibration_failed_beep'
+                doBeep=el.feedbackbeep;
+                f=el.calibration_failed_beep(1);
+                v=el.calibration_failed_beep(2);
+                d=el.calibration_failed_beep(3);
+            case 'calibration_success_beep'
+                doBeep=el.feedbackbeep;
+                f=el.calibration_success_beep(1);
+                v=el.calibration_success_beep(2);
+                d=el.calibration_success_beep(3);
+            case 'drift_correction_failed_beep'
+                doBeep=el.feedbackbeep;
+                f=el.drift_correction_failed_beep(1);
+                v=el.drift_correction_failed_beep(2);
+                d=el.drift_correction_failed_beep(3);
+            case 'drift_correction_success_beep'
+                doBeep=el.feedbackbeep;
+                f=el.drift_correction_success_beep(1);
+                v=el.drift_correction_success_beep(2);
+                d=el.drift_correction_success_beep(3);
+            otherwise
+                % some defaults
+                doBeep=el.feedbackbeep;
+                f=500;
+                v=0.5;
+                d=1.5;
+        end
+        
+        if doBeep==1
+            if PsychPortAudio('GetOpenDeviceCount') > 0
```

Comment:
This is not an optimal approach. See `help Snd` section about pahandle sharing and interop with Eyelink for a better one.

Reply:
resolved in 31a44c3bca8ed8ae93777eb1b84fd8f6f12256d5 with proper support for PsychPortAudio